dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/v2/DataUploadRunnable.kt

```diff
@@ -54,28 +55,40 @@ internal class DataUploadRunnable(
             val context = contextProvider.context
             // TODO RUMM-0000 it should be already on the worker thread and if readNextBatch is async,
             //  we should wait until it completes before scheduling further
-            val lock = CountDownLatch(1)
-            storage.readNextBatch(
-                noBatchCallback = {
-                    increaseInterval()
-                    lock.countDown()
+            var batchesSent = 0
+            var lastUploadWasSuccess = false
+            do  {
+                val lock = CountDownLatch(1)
+                storage.readNextBatch(
+                    noBatchCallback = {
+                        increaseInterval()
+                        lock.countDown()
+                        batchesSent = maxBatchesPersUpload
+                        lastUploadWasSuccess = false
+                    }
+                ) { batchId, reader ->
+                    try {
+                        val batch = reader.read()
+                        val batchMeta = reader.currentMetadata()
+
+                        val status = consumeBatch(
```

Comment:
are we ok that each call to `consumeBatch` will decrease scheduling interval in case of upload success? Should we decrease interval only when the whole queue is processed?